### PR TITLE
Cache interpreter info in memory

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: crates/pet-core/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,6 @@ msvc_spectre_libs = { version = "0.1.1", features = ["error"] }
 
 [dependencies]
 pet-fs = { path = "../pet-fs" }
-pet-python-utils = { path = "../pet-python-utils" }
```
Comment (DonJayamanne) [inline crates/pet-core/Cargo.toml:11]: Had to remove this incorrect depenedency, core should not have any other dependencies related to python interpreters and the like.
Hence the large number of files touched.

File: crates/pet-core/src/env.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::path::PathBuf;
```
Comment (DonJayamanne) [inline crates/pet-core/src/env.rs:4]: Moved from `pet-python-utils`

File: crates/pet-python-utils/src/cache.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            generate_hash(&PathBuf::from(
+                "/Users/donjayamanne/DEMO/.venvTestInstall1/bin/python3.12"
+            )),
+            "ecb0ee73d6ddfe97"
+        );
+    }
+
+    // #[test]
+    // #[cfg(unix)]
+    // fn test_hash_generation_upper_case() {
```
Comment (DonJayamanne) [inline crates/pet-python-utils/src/cache.rs:134]: Will be enabled/implemented in a separate PR.
This pr will only focus on in memory caching.

File: crates/pet-python-utils/src/env.rs [hunk 4]
```diff
@@ -2,63 +2,14 @@
...
-    pub version: Option<String>,
-    /// Possible symlink (or known alternative link).
-    /// For instance:
-    ///
-    /// If `executable` is `/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/bin/python3.12`,
-    /// then `symlink`` can be `/opt/homebrew/bin/python3.12` (or vice versa).
-    pub symlinks: Option<Vec<PathBuf>>,
-}
-
-impl PythonEnv {
```
Comment (DonJayamanne) [inline crates/pet-python-utils/src/env.rs:35]: Moved to `pet-core`

**Review by alexr00 — APPROVED** [review]

**Pull request merged**